bundler/index.js

```diff
@@ -4,70 +4,115 @@ const HtmlParser = require('./modules/htmlParser');
+    pathConfig,
+    tagConfig,
   ) {
-    this.fileManager = fileManager;
-    this.uglifier = uglifier;
-    this.htmlParser = htmlParser;
+    this.pathConfig = pathConfig;
+    this.tagConfig = tagConfig;
+    this.fileManager = new FileManager();
+    this.uglifier = new Uglifier();
+    this.htmlParser = new HtmlParser(tagConfig);
   }
 
-  async createBundle(htmlFilePath, schema = null) {
-    if (!schema) {
-      schema = this.createSchema(htmlFilePath);
+  async create() {
+    try {
+      this._logBundleStarting();
+      this._dist();
+      this._assets();
+      const htmlPath = this.pathConfig.HTML;
+      const htmlData = await this.fileManager.readAggregateFilesData(htmlPath);
+      if (!htmlData) {
+        this._logFileNotCreated(htmlPath);
+        throw `File not created ${htmlPath}`;
+      }
+      const cssBundleName = await this._css(htmlData);
+      const jsBundleName = await this._js(htmlData);
+      await this._html(htmlData, htmlPath, cssBundleName, jsBundleName);
+      this._logBundleFinished();
+    } catch(error) {
+      this._removeDistFolder();
+      this._logBundleFailed();
     }
-    // html
-    await this.processBundleUnit(
-      schema.HTML.input,
-      schema.HTML.output,
-      this.uglifier.uglifyHtml,
-      true
-    );
-    // css
-    await this.processBundleUnit(
-      schema.CSS.input,
-      schema.CSS.output,
-      this.uglifier.uglifyCss
-    );
-    // js
-    await this.processBundleUnit(
-      schema.JS.input,
-      schema.JS.output,
-      this.uglifier.uglifyJs
-    );
-  }
-
-  createSchema(htmlFilePath) {
-    const rawPathes = this.htmlParser.findPathesInHtml(htmlFilePath);
-    
-    const pathes = {};
-    Object.keys(rawPathes).forEach((key) => {
-      pathes[key] = {
-        input: rawPathes[key],
-        output: `dist/bundle.${key.toLowerCase()}`
-      };
-    });
-
-    const split = htmlFilePath.split('/');
-    const htmlFileName = split[split.length - 1];
-    pathes.HTML = {
-      input: [ htmlFilePath ],
-      output: `dist/${htmlFileName}`
-    };
+  }
+
+  _logBundleStarting() {
+    console.log(`⌛ Starting Bundle for ${this.pathConfig.HTML}`)
+  }
+
+  _logFileNotCreated(filePath) {
+    console.log(`🚫 ${filePath}`);
+  }
+
+  _logFileNotFound(filePath) {
+    console.log(`❓ ${filePath}`);
+  }
 
-    return pathes;
+  _logFileCreated(filePath) {
+    console.log(`📁 ${filePath} ✓`)
   }
 
-  async processBundleUnit(filesPathsList, outputPath, uglifyMethod, isHtml = false) {
-    let data = await this.fileManager.readAggregateFilesData(filesPathsList);
-    if (isHtml) {
-      data = this.htmlParser.replaceSources(data);
+  _logBundleFinished() {
+    console.log(`✅ Finished Bundle for ${this.pathConfig.HTML} → ${this.pathConfig.DIST}`)
+  }
+
+  _logBundleFailed() {
+    console.log(`❌ Failed Bundle ${this.pathConfig.HTML}`)
+  }
+
+  _removeDistFolder() {
+    this.fileManager.removeFolder(this.pathConfig.DIST);
+  }
+
```

Comment:
It doesn't make code bad. It's more a matter of taste. But I'm a big fan of having the method names that tell what exactly methods do.

How do you think, will it be more understandable to read the code if we will rename:
`_write => _writeToFile`,
`_dist => _setupDistFolder`,
`_src => "_uglifySrc" or "_processSrc" or "_handleSrc"`,
`_assets => _copyAssets`
?